lasagne/layers/special.py

```diff
@@ -544,6 +546,296 @@ def _meshgrid(height, width):
     return grid
 
 
+class TPSTransformerLayer(MergeLayer):
+    """
+    Spatial transformer layer
+
+    The layer applies a thin plate spline transformation [2]_ on the input
+    as in [1]_. The thin plate spline transform is determined based on the
+    movement of some number of control points. The starting positions for
+    these control points are fixed. The output is interpolated with a
+    bilinear transformation.
+
+    Parameters
+    ----------
+    incoming : a :class:`Layer` instance or a tuple
+        The layer feeding into this layer, or the expected input shape. The
+        output of this layer should be a 4D tensor, with shape
+        ``(batch_size, num_input_channels, input_rows, input_columns)``.
+
+    localization_network : a :class:`Layer` instance
+        The network that calculates the parameters of the thin plate spline
+        transformation as the x and y coordinates of the destinations of each
+        control point. The output of the localization network must be twice
+        the number of control points.
```

Comment:
Maybe a little more explicit: The output of the localization network must be a 2D tensor, with shape ``(batch_size, 2 * num_control_points)``.